packages/fetch-client/src/http-client.ts

```diff
@@ -327,11 +328,12 @@ export class HttpClient {
   }
 }
 
-function parseHeaderValues(headers: object): object {
-  const parsedHeaders = {};
-  for (const name in headers || {}) {
-    if (headers.hasOwnProperty(name)) {
-      parsedHeaders[name] = (typeof headers[name] === 'function') ? headers[name]() : headers[name];
+function parseHeaderValues(headers: Record<string, unknown> | undefined): Record<string, string>  {
+  const parsedHeaders: Record<string, string> = {};
+  const $headers = headers !== undefined ? headers : {};
+  for (const name in $headers) {
+    if ($headers.hasOwnProperty(name)) {
+      parsedHeaders[name] = (typeof $headers[name] === 'function') ? ($headers[name] as () => string)() : $headers[name] as string;
```

Comment:
```suggestion
      parsedHeaders[name] = (typeof $headers[name] === 'function')
        ? ($headers[name] as () => string)()
        : $headers[name] as string;
``` 

<a id="85cd8ef0-386e-11e9-873b-15af3aec25b1" href="none" codereview-uuid-reserved-tag />